src/riscv/lib/src/machine_state/memory.rs

```diff
@@ -307,6 +310,67 @@ pub trait MemoryConfig: Send + 'static {
         F: FnManager<Ref<'a, M>>;
 }
 
+/// Can be pushed to a stack frame.
+/// Ensures that elements are pushed onto a stack frame in the correct order.
+///
+/// See also [StackFramePop]
+pub trait StackFramePush: Default + Sized {
+    type Above: StackFramePop;
+    const ALIGN: u64 = 8;
```

Comment:
```suggestion

    const _ALIGN_CHECK: () = {
        if ALIGN as usize % std::mem::size_of::<<Self::Above as StackFramePop>::Data>() != 0 {
            panic!("Stack frame alignment must be a multiple of 8 bytes");
        };
    };
```

It doesn't look like implementors specialise this constant. And because currently it has to be the same as the one for `StackFramePop` why not have it as a module-level constant?

You can add a compile-time verification the `ALIGN` is a multiple of the data you are using instead.

Alternatively, if the alignment can be different for separate items, can [`align_of`](https://doc.rust-lang.org/stable/std/mem/fn.align_of.html) be used inline directly?

Speaking of alignments, is there a requirement for the stack pointer alignment in general, besides alignment for a simple access? (I haven't read the spec in too much detail lately)